File: app/src/main/java/de/xikolo/utils/DisplayUtil.java
package de.xikolo.utils;

import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Build;
import android.util.DisplayMetrics;

public class DisplayUtil {

    public static boolean is7inchTablet(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();

        float dpHeight = displayMetrics.heightPixels / displayMetrics.density;
        float dpWidth = displayMetrics.widthPixels / displayMetrics.density;

        return dpHeight >= 600 || dpWidth >= 600;
    }

    public static boolean is10inchTablet(Context context) {
        DisplayMetrics displayMetrics = context.getResources().getDisplayMetrics();

        float dpHeight = displayMetrics.heightPixels / displayMetrics.density;
        float dpWidth = displayMetrics.widthPixels / displayMetrics.density;

        return dpHeight >= 720 || dpWidth >= 720;
    }

    public static boolean supportsPictureInPicture(Context context) {
        return Build.VERSION.SDK_INT >= 26 && context.getPackageManager().hasSystemFeature(PackageManager.FEATURE_PICTURE_IN_PICTURE);
    }

}

File: app/src/main/java/de/xikolo/utils/DisplayUtil.kt
package de.xikolo.utils

import android.app.Activity
import android.content.Context
import android.content.res.Configuration
import android.graphics.Point
import android.os.Build
import android.util.DisplayMetrics

object DisplayUtil {

    @JvmStatic
    fun is7inchTablet(context: Context): Boolean {
        val displayMetrics = context.resources.displayMetrics

        val dpHeight = displayMetrics.heightPixels / displayMetrics.density
        val dpWidth = displayMetrics.widthPixels / displayMetrics.density

        return dpHeight >= 600 || dpWidth >= 600
    }

    @JvmStatic
    fun is10inchTablet(context: Context): Boolean {
        val displayMetrics = context.resources.displayMetrics

        val dpHeight = displayMetrics.heightPixels / displayMetrics.density
        val dpWidth = displayMetrics.widthPixels / displayMetrics.density

        return dpHeight >= 720 || dpWidth >= 720
    }

    fun getVideoThumbnailSize(activity: Activity): Point {
        var size = Point()
        activity.windowManager.defaultDisplay.getSize(size)
        size = if (activity.resources.configuration.orientation == Configuration.ORIENTATION_PORTRAIT) {
            Point(size.x, size.x / 16 * 9)
        } else {
            Point((size.x * 0.6).toInt(), (size.x * 0.6 / 16 * 9).toInt())
        }
        return size
    }

    fun getAspectRatio(activity: Activity): Float {
        val displayMetrics = DisplayMetrics()
        activity.windowManager.defaultDisplay.getRealMetrics(displayMetrics)
        return if(activity.resources?.configuration?.orientation == Configuration.ORIENTATION_PORTRAIT) {
            displayMetrics.heightPixels.toFloat() / displayMetrics.widthPixels
        } else {
            displayMetrics.widthPixels.toFloat() / displayMetrics.heightPixels
        }
    }

    fun hasDisplayCutouts(activity: Activity): Boolean {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.P && activity.window.decorView.rootWindowInsets.displayCutout != null
    }

}
